admiral/envs/components/examples/bird_flight.py

```diff
@@ -0,0 +1,111 @@
+
+from matplotlib import pyplot as plt
+import numpy as np
+
+from admiral.envs.components.agent import PositionAgent, SpeedAngleAgent
+from admiral.envs.components.state import ContinuousPositionState, SpeedAngleState
+from admiral.envs.components.actor import SpeedAngleMovementActor
+from admiral.envs.components.observer import SpeedObserver, AngleObserver
+from admiral.envs.components.done import TooCloseDone
+from admiral.envs import AgentBasedSimulation
+from admiral.tools.matplotlib_utils import mscatter
+
+class BirdAgent(PositionAgent, SpeedAngleAgent): pass
+
+class Flight(AgentBasedSimulation):
+    def __init__(self, **kwargs):
+        self.agents = kwargs['agents']
+
+        # State
+        self.position = ContinuousPositionState(**kwargs)
+        self.speed_angle = SpeedAngleState(**kwargs)
+
+        # Actor
+        self.move_actor = SpeedAngleMovementActor(position=self.position, speed_angle=self.speed_angle, **kwargs)
+
+        # Observer
+        self.speed_observer = SpeedObserver(**kwargs)
+        self.angle_observer = AngleObserver(**kwargs)
+
+        # Done
+        self.done = TooCloseDone(position=self.position, **kwargs)
+
+        from gym.spaces import MultiBinary
+        for agent in self.agents.values():
+            agent.observation_space = {'dummy': MultiBinary(1)}
+        self.finalize()
+
+    def reset(self, **kwargs):
+        self.position.reset(**kwargs)
+        self.speed_angle.reset(**kwargs)
+    
+    def step(self, action_dict, **kwargs):
+        for agent, action in action_dict.items():
+            self.move_actor.move(self.agents[agent], action['accelerate'][0], action['banking_angle_change'][0], **kwargs)
+        
+    def render(self, fig=None, **kwargs):
+        fig.clear()
+
+        # Draw the resources
+        ax = fig.gca()
+
+        # Draw the agents
+        ax.set(xlim=(0, self.position.region), ylim=(0, self.position.region))
+        ax.set_xticks(np.arange(0, self.position.region, 1))
+        ax.set_yticks(np.arange(0, self.position.region, 1))
+
+        agents_x = [agent.position[1] + 0.5 for agent in self.agents.values()]
```

Comment:
This is inconsistent with the way we have defined the movements in the speedangle actor. There, we consider 0 to be y and 1 to be x. See note below and treat them consistently.